src/objects/texts/zcl_abapgit_lxe_texts.clas.abap

```diff
@@ -0,0 +1,169 @@
+          iv_object_type     TYPE trobjtype
+        RETURNING
+          VALUE(rt_obj_list) TYPE lxe_tt_colob,
+      get_lxe_texts
+        IMPORTING
+                  iv_original_language TYPE spras
+                  iv_obj_name          TYPE sobj_name
+                  iv_object_type       TYPE trobjtype
+        RETURNING VALUE(rt_lxe_texts)  TYPE zif_abapgit_lxe_texts~ty_tlxe_i18n,
+      deserialize_lxe_texts
+        IMPORTING
+          it_lxe_texts TYPE zif_abapgit_lxe_texts~ty_tlxe_i18n,
+      get_installed_languages
+        RETURNING
+          VALUE(rt_installed_languages) TYPE zif_abapgit_lxe_texts~ty_languages.
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_lxe_texts IMPLEMENTATION.
+  METHOD get_lang_iso4.
+    CALL FUNCTION 'LXE_T002_CONVERT_2_TO_4'
+      EXPORTING
+        old_r3_lang = iv_src
+      IMPORTING
+        new_lang    = rv_iso4.
+  ENDMETHOD.
+
+
+  METHOD get_lxe_object_list.
+
+
+    DATA lv_object_name TYPE trobj_name.
+
+    lv_object_name = iv_obj_name.
+    CALL FUNCTION 'LXE_OBJ_EXPAND_TRANSPORT_OBJ'
+      EXPORTING
+        pgmid    = 'R3TR'
+        object   = iv_object_type
+        obj_name = lv_object_name
+      TABLES
+        ex_colob = rt_obj_list.
+
+  ENDMETHOD.
+  METHOD get_lxe_texts.
+    DATA:
+      lt_obj_list            TYPE lxe_tt_colob,
+      lt_installed_languages TYPE TABLE OF langu,
+
+      ls_lxe_text_item       TYPE zif_abapgit_lxe_texts~ty_lxe_i18n.
+
+    FIELD-SYMBOLS:
+      <lv_language>   TYPE langu,
+      <lv_lxe_object> TYPE lxe_colob.
+
+    lt_obj_list = get_lxe_object_list(
+                    iv_obj_name    = iv_obj_name
+                    iv_object_type = iv_object_type ).
+
+    lt_installed_languages = get_installed_languages( ).
+
+    LOOP AT lt_obj_list ASSIGNING <lv_lxe_object>.
+      CLEAR ls_lxe_text_item.
+      ls_lxe_text_item-custmnr = <lv_lxe_object>-custmnr.
+      ls_lxe_text_item-objtype = <lv_lxe_object>-objtype.
+      ls_lxe_text_item-objname = <lv_lxe_object>-objname.
+
+      LOOP AT lt_installed_languages ASSIGNING <lv_language>.
+        ls_lxe_text_item-source_lang = get_lang_iso4( iv_original_language ).
+        ls_lxe_text_item-target_lang = get_lang_iso4( <lv_language> ).
+        IF ls_lxe_text_item-source_lang = ls_lxe_text_item-target_lang.
+          CONTINUE. " if source = target -> skip
+        ENDIF.
+
+        CLEAR ls_lxe_text_item-text_pairs.
+        CALL FUNCTION 'LXE_OBJ_TEXT_PAIR_READ'
+          EXPORTING
+            s_lang    = ls_lxe_text_item-source_lang
+            t_lang    = ls_lxe_text_item-target_lang
+            custmnr   = ls_lxe_text_item-custmnr
+            objtype   = ls_lxe_text_item-objtype
+            objname   = ls_lxe_text_item-objname
+          TABLES
+            lt_pcx_s1 = ls_lxe_text_item-text_pairs.
+
+        DELETE ls_lxe_text_item-text_pairs WHERE t_text IS INITIAL. " No Target Text, no translation to be transported
```

Comment:
do we need to keep these entries to overwrite previous translations that were replaced by a blank text?

Reply:
I think you are right, so I removed the deletion. Of course that will lead to more load in the XML in the ALL_INSTALLED_LANGUAGES scenario, which is what I wanted to prevent. But I think it is acceptable.
Anyway during a quick test I already saw that the XML will not contain the empty target text. Probably, the XML writing code will not add empty tags?
Not sure if it causes problems, did not have time to test the deserialization part, my time slot for this project for today is exceeded.